app/models/proposal.rb

```diff
@@ -107,18 +108,46 @@ def remove_approver(email)
     approval.destroy
   end
 
-  def initialize_approvals()
-    if self.linear? && self.approvals.any?
-      self.approvals.update_all(status: 'pending')
-      self.approvals.first.make_actionable!
-    elsif self.parallel?
-      self.approvals.update_all(status: 'actionable')
+  # Set the approver list, from any start state
+  def approvers=(approver_list)
+    approvals = approver_list.each_with_index.map do |approver, idx|
+      approval = self.approval_for(approver)
+      approval ||= Approval.new(user: approver, proposal: self)
+      approval.position = idx + 1   # start with 1
+      approval
+    end
+    self.approvals = approvals
+    self.kickstart_approvals()
+    self.reset_status()
+  end
```

Comment:
Feels a little dirty to override a built-in accessor, even though it throws an error since this is a `through` relation. Maybe just make a note of it?